src/include/sof/atomic.h

```diff
@@ -11,6 +11,8 @@
 #include <arch/atomic.h>
 #include <stdint.h>
 
+#ifndef __ZEPHYR__
```

Comment:
wouldn't it be better and more correct to move this `#ifndef` 3 lines up and to `#include <sys/atomic.h>` on the `#else` branch? Or better invert the condition to `#ifdef` to avoid a negation.

Reply:
Agree